src/utils/utils.tsx

```diff
@@ -10,14 +10,45 @@ export enum ElementType {
   USER = "users",
   DATACENTER = "datacenters"
 }
-export interface AssetObject extends ElementObject {
+export enum PowerSide {
+  LEFT = "L",
+  RIGHT = "R"
+}
+export interface AssetObjectOld extends ElementObject {
   hostname: string;
   rack_position: string;
   model: ModelObjectOld;
   rack: RackObject;
   owner?: string;
   comment?: string;
 }
+
+export interface ShallowAssetObjectOld extends ElementObject {
+  hostname: string;
+  rack_position: string;
+  model?: string;
+  rack?: string;
+  owner?: string;
+  comment?: string;
+}
+export interface AssetObject extends ParentAssetObject {
+  model: ModelObject;
+  rack: RackObject;
+}
+
+interface ParentAssetObject extends ElementObject {
+  hostname: string;
+  rack_position: string;
+  mac_addresses: { [port: string]: string };
```

Comment:
What is this syntax? Mostly just asking out of curiosity, I've never seen it before

Reply:
its basically a dict syntax, so mac addresses has keys of type string, and values of type string.